shpool/tests/detach.rs

```diff
@@ -34,6 +34,9 @@ fn single_running() -> anyhow::Result<()> {
 
         daemon_proc.events = Some(waiter.wait_final_event("daemon-bidi-stream-done")?);
 
+        let attach_exit_status = attach_proc.proc.wait()?;
+        assert!(attach_exit_status.success());
```

Comment:
dumb question (to save me time), this looks to return an unwraped i32. where does the `.success()` method come from on this type?
If we're using traits somehow, should we not define an i32 type but rather make a new type?
I may have gotten all this wrong. 

for this comment, it is ok if `.success()` means we have exit status 0 and not other way around.
(ditto for another similar line, given this is only two lines i wont add a ccmment to it)

Reply:
The stdlib has an `ExitStatus` type that has a few methods hanging off of it. You can unwrap to the inner int if you want, but for just checking success you can use the methods.

Reply:
im seeing the return type as `Option<int32>` though, is that correct?

here: https://github.com/shell-pool/shpool/blob/55be263b397ad3cebed9e7a7d0d413603bc16a6f/libshpool/src/daemon/exit_notify.rs#L40